### PR TITLE
Add tests to improve coverage

### PR DESCRIPTION
**Pull request by kleinschmidt**
This adds tests for 

* `implicit_intercept` and `drop_intercept` model traits
* `has_schema`

And fixes at least one bug that turned up in the process (a missing method for
`has_schema(::MatrixTerm)`).

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
Can't really object to a PR that just adds tests :stuck_out_tongue:

File: test/modelmatrix.jl [hunk 1]
```diff
@@ -350,6 +350,26 @@
         @test reduce(vcat, last.(modelcols.(Ref(f), Tables.rowtable(d)))') == modelmatrix(f,d)
     end
 
+    @testset "modelmatrix and response set schema if needed" begin
+        using StatsModels: has_schema
+
```
Comment (nalimilan) [inline test/modelmatrix.jl:355]:
```suggestion
```

File: test/traits.jl [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        ff = apply_schema(f, sch, DroppyMod)
+        @test !hasintercept(ff)
+        @test !omitsintercept(ff)
+        ff0 = apply_schema(f0, sch, DroppyMod)
+        @test !hasintercept(ff0)
+        @test omitsintercept(ff0)
+
+        @test drop_intercept(DroppyMod()) == drop_intercept(DroppyMod)
+        # drop_intercept blocks implicit_intercept == true
+        @test implicit_intercept(DroppyMod) == true
```
Comment (nalimilan) [inline test/traits.jl:39]:
```suggestion
        @test implicit_intercept(DroppyMod)
```

**Review by nalimilan — APPROVED** [review]
LGTM too. Just two small things.

**Pull request merged**